osutil/disks/disks_linux.go

```diff
@@ -251,6 +251,27 @@ func DiskFromMountPoint(mountpoint string, opts *Options) (Disk, error) {
 	return diskFromMountPoint(mountpoint, opts)
 }
 
+// DiskFromPartitionDeviceNode finds a matching Disk that the specified
+// partition node resides on.
+func DiskFromPartitionDeviceNode(node string) (Disk, error) {
+	// TODO: support options such as IsDecryptedDevice
+	return diskFromPartitionDeviceNode(node)
+}
+
+var diskFromPartitionDeviceNode = func(node string) (Disk, error) {
+	// get the udev properties for this device node
+	props, err := udevPropertiesForName(node)
+	if err != nil {
+		return nil, err
+	}
+
+	disk, err := diskFromPartUdevPropsAndOpts(props, nil)
```

Comment:
minor: we mostly write `UDev` vs `Udev` (532 vs 127 :-) ).

Reply:
I would also remove the `AndOpts` part from the name

Reply:
agree, let's try to be consistent about casing. Also agree that most of our functions taking opts don't spell that out in the name

Reply:
done